src/java/org/apache/cassandra/db/lifecycle/Tracker.java

```diff
@@ -269,13 +271,33 @@ public void reset(Memtable memtable)
                           SSTableIntervalTree.empty()));
     }
 
-    public Throwable dropSSTablesIfInvalid(Throwable accumulate)
+    public Throwable dropOrUnloadSSTablesIfInvalid(String message, @Nullable Throwable accumulate)
     {
         if (!isDummy() && !cfstore.isValid())
-            accumulate = dropSSTables(accumulate);
+            accumulate = dropOrUnloadSSTables(cfstore.status(), message, accumulate);
         return accumulate;
     }
 
+    /**
+     * Drop sstables if cfs is invalidated with drop data; otherwise unload sstables
+     *
+     * @param message for logging
+     * @return accumulated throwable
+     */
+    public Throwable dropOrUnloadSSTables(ColumnFamilyStore.STATUS status, String message, @Nullable Throwable accumulate)
+    {
+        if (status.isInvalidAndShouldDropData())
```

Comment:
In fact, it's only used in one place - it could be `private`, but I think it will be more clear just to inline the code; I don't really see a reason to make is a separate method in this case.